src/blueteam/Robot.java

```diff
@@ -328,4 +343,125 @@ void dodge() {
 	}
+
+	/**
+	 *
+	 * @param map
+	 *            A MapLocation to convert to integer representation
+	 * @return An array arr such that: arr[0] - integer part of x arr[1] -
+	 *         decimal part of x * 10^6 and rounded arr[2] - integer part of y
+	 *         arr[3] - decimal part of y * 10^6 and rounded
+	 */
+	int[] convertMapLocation(MapLocation map) {
+		float xcoord = map.x;
+		float ycoord = map.y;
+		int[] returnarray = new int[4];
+		returnarray[0] = Math.round(xcoord - (xcoord % 1));
+		returnarray[1] = Math.toIntExact(Math.round((xcoord % 1) * Math.pow(10, 6)));
+		returnarray[2] = Math.round(ycoord - (ycoord % 1));
+		returnarray[3] = Math.toIntExact(Math.round((ycoord % 1) * Math.pow(10, 6)));
+		return (returnarray);
+	}
+
+	/**
+	 *
+	 * @param arr
+	 *            An array arr such that: arr[0] - integer part of x arr[1] -
+	 *            decimal part of x * 10^6 and rounded arr[2] - integer part of
+	 *            y arr[3] - decimal part of y * 10^6 and rounded
+	 * @return A MapLocation instantiated from the coordinates given by array
+	 */
+	MapLocation convertLocationInts(int[] arr) {
+		float xcoord = (float) (arr[0] + arr[1] / Math.pow(10, 6));
+		float ycoord = (float) (arr[2] + arr[3] / Math.pow(10, 6));
+		return (new MapLocation(xcoord, ycoord));
+	}
+
+	MapLocation readLocation(int firstChannel) throws GameActionException {
+		int[] array = new int[4];
+		array[0] = rc.readBroadcast(firstChannel);
+		array[1] = rc.readBroadcast(firstChannel + 1);
+		array[2] = rc.readBroadcast(firstChannel + 2);
+		array[3] = rc.readBroadcast(firstChannel + 3);
+		return convertLocationInts(array);
+	}
+
+	void writeLocation(MapLocation map, int firstChannel) throws GameActionException {
+		int[] arr = convertMapLocation(map);
+		rc.broadcast(firstChannel, arr[0]);
+		rc.broadcast(firstChannel + 1, arr[1]);
+		rc.broadcast(firstChannel + 2, arr[2]);
+		rc.broadcast(firstChannel + 3, arr[3]);
+	}
+
+	void wander() {
+		Direction dir = randomFreeDirection();
+		tryMove(dir);
+	}
+
+	boolean slugMoveToTarget(MapLocation target, float strideRadius) {
+
+		// when trying to move, let's look forward, then incrementing left and
+		// right.
+		float[] toTry = { 0, (float) Math.PI / 4, (float) -Math.PI / 4, (float) Math.PI / 2, (float) -Math.PI / 2,
+				3 * (float) Math.PI / 4, -3 * (float) Math.PI / 4, -(float) Math.PI };
+
+		MapLocation ourLoc = rc.getLocation();
+		Direction toMove = ourLoc.directionTo(target);
+
+		// let's try to find a place to move!
+		for (int i = 0; i < toTry.length; i++) {
+			Direction dirToTry = toMove.rotateRightDegrees(toTry[i]);
+			if (rc.canMove(dirToTry, strideRadius)) {
+				// if that location is free, let's see if we've already moved
+				// there before (aka, it's in our tail)
+				MapLocation newLocation = ourLoc.add(dirToTry, strideRadius);
+				boolean haveWeMovedThereBefore = false;
+				for (MapLocation loc : oldLocations) {
+					if (newLocation.distanceTo(loc) < strideRadius * strideRadius) {
```

Comment:
BTW, can you explain why there is the strideRadius squared? I'd expect either only strideRadius or distanceSquaredTo...